cli/functionary/tokens.py

```diff
@@ -1,25 +1,9 @@
-import json
+from .client import post
 
-import click
-import requests
 
-
-def login(login_url: str, user: str, password: str):
-
-    try:
-        login_response = requests.post(
-            f"{login_url}", data={"username": user, "password": password}
-        )
-        # check status code/message on return then exit
-        if login_response.ok:
-            token = json.loads(login_response.text).get("token")
-            return token
-        else:
-            raise click.ClickException(
-                f"Failed to login: {login_response.status_code}\n"
-                f"Response: {login_response.text}"
-            )
-    except requests.ConnectionError:
-        raise click.ClickException(f"Unable to connect to {login_url}.")
-    except requests.Timeout:
-        raise click.ClickException("Timeout occurred waiting for login")
+def login(user: str, password: str):
+    login_response = post(
+        "api-token-auth", data={"username": user, "password": password}
+    )
+    token = login_response.get("token")
```

Comment:
The most super minor thing, but I notice this pattern a lot of setting a variable, only to return it on the very next line.  There's nothing wrong with that really, but you can just return things directly:

```python
return login_response.get("token")
```